File: src/main/java/io/zeitwert/fm/dms/model/enums/CodeContentKind.java
package io.zeitwert.fm.dms.model.enums;

import lombok.Data;
import lombok.experimental.SuperBuilder;

import java.util.List;

import io.dddrive.enums.model.base.EnumeratedBase;

@Data
@SuperBuilder
public final class CodeContentKind extends EnumeratedBase {

	@Override
	public boolean equals(Object other) {
		return super.equals(other);
	}

	@Override
	public int hashCode() {
		return super.hashCode();
	}

	public List<CodeContentType> getContentTypes() {
		return CodeContentTypeEnum.getContentTypes(this);
	}

	public List<String> getExtensions() {
		return this.getContentTypes().stream().map(ct -> "." + ct.getExtension()).toList();
	}

}

File: fm-domain/src/main/java/io/zeitwert/fm/dms/model/enums/CodeContentKind.kt
package io.zeitwert.fm.dms.model.enums

import dddrive.ddd.model.EnumeratedEnum
import dddrive.ddd.model.base.EnumerationBase

enum class CodeContentKind(
	override val defaultName: String,
) : EnumeratedEnum {

	DOCUMENT("Document"),
	FOTO("Foto"),
	VIDEO("Video"),
	;

	override val enumeration get() = Enumeration

	fun getContentTypes(): List<CodeContentType> = CodeContentType.entries.filter { it.contentKind == this }

	fun getExtensions(): List<String> = getContentTypes().map { ".${it.extension}" }

	companion object Enumeration : EnumerationBase<CodeContentKind>(CodeContentKind::class.java) {
		init {
			entries.forEach { addItem(it) }
		}

		@JvmStatic
		fun getContentKind(itemId: String?) = if (itemId != null) getItem(itemId) else null
	}
}
